pkg/zack/layout_split.go

```diff
@@ -0,0 +1,254 @@
+			startHeight = int(m.fixedStartSize)
+			endHeight = bounds.Height - startHeight
+		} else if m.fixedEndSize > 0 {
+			endHeight = int(m.fixedEndSize)
+			startHeight = bounds.Height - endHeight
+		} else {
+			startHeight = int(float64(bounds.Height) * m.fraction)
+			endHeight = bounds.Height - startHeight
+		}
+		m.setStartChildBounds(NewBounds(bounds.Width, startHeight))
+		m.setEndChildBounds(NewBounds(bounds.Width, endHeight))
+		return
+	}
+
+	log.Warn().Msgf("(SplitLayout.SetBounds) Invalid SplitLayoutModel orientation: %v", m.orientation)
+}
+
+// refreshBounds send the newest bounds to the start and end child boundables.
+// This is useful when the _layout_ itself changes, e.g. when you increase or decrease the fraction,
+func (m *SplitLayoutModel) refreshBounds() {
+	m.SetBounds(m.layoutBounds)
+}
+
+// setStartChildBounds sets the bounds for the start child view and updates its boundable.
+func (m *SplitLayoutModel) setStartChildBounds(bounds Bounds) {
+	m.startChildBounds = bounds
+	if !bounds.IsValid() {
+		log.Warn().Msgf("(SplitLayout.setStartChildBounds) start child bounds are invalid: %v", bounds)
+		return
+	}
+	if m.startChildBoundable != nil {
+		m.startChildBoundable.SetBounds(bounds)
+	}
+}
+
+// setEndChildBounds sets the bounds for the end child view and updates its boundable.
+func (m *SplitLayoutModel) setEndChildBounds(bounds Bounds) {
+	m.endChildBounds = bounds
+	if !bounds.IsValid() {
+		log.Warn().Msgf("(SplitLayout.setEndChildBounds) end child bounds are invalid: %v", bounds)
+		return
+	}
+	if m.endChildBoundable != nil {
+		m.endChildBoundable.SetBounds(bounds)
+	}
+}
+
+func (m *SplitLayoutModel) SetFraction(fraction float64) {
+	m.fraction = Clamp(fraction, 0, 1)
+	m.refreshBounds()
+}
+
+// Increase increases the fraction of the start view by 1/width
+func (m *SplitLayoutModel) Increase() {
+	switch m.orientation {
+	case Horizontal:
+		m.SetFraction(m.fraction + (1.0 / float64(m.layoutBounds.Width)))
+	case Vertical:
+		m.SetFraction(m.fraction + (1.0 / float64(m.layoutBounds.Height)))
+	default:
+		log.Warn().Msgf("(SplitLayout.Increase) Invalid SplitLayoutModel orientation: %v", m.orientation)
+	}
+}
+
+// Decrease decreases the fraction of the start view by 1/width
+func (m *SplitLayoutModel) Decrease() {
+	switch m.orientation {
+	case Horizontal:
+		m.SetFraction(m.fraction - (1.0 / float64(m.layoutBounds.Width)))
+	case Vertical:
+		m.SetFraction(m.fraction - (1.0 / float64(m.layoutBounds.Height)))
+	default:
+		log.Warn().Msgf("(SplitLayout.Decrease) Invalid SplitLayoutModel orientation: %v", m.orientation)
+	}
+}
+
+// ToggleOrientation toggles the orientation of the layout between horizontal and vertical
+func (m *SplitLayoutModel) ToggleOrientation() {
+	if m.orientation == Horizontal {
+		m.orientation = Vertical
+	} else {
+		m.orientation = Horizontal
+	}
+	// we need to refresh the bounds here since the orientation changes -> the bounds of the children will change
+	m.refreshBounds()
```

Comment:
Don't we also need to re-render here? Or should this be handled by the application logic?